l1-contracts/contracts/state-transition/chain-deps/facets/Executor.sol

```diff
@@ -259,16 +281,47 @@ contract ExecutorFacet is ZKChainBase, IExecutor {
             }
         }
 
-        // We only require 7 logs to be checked, the 8th is if we are expecting a protocol upgrade
-        // Without the protocol upgrade we expect 7 logs: 2^7 - 1 = 127
-        // With the protocol upgrade we expect 8 logs: 2^8 - 1 = 255
+        // We only require 8 logs to be checked, the 9th is if we are expecting a protocol upgrade
+        // Without the protocol upgrade we expect 8 logs: 2^8 - 1 = 255
+        // With the protocol upgrade we expect 9 logs: 2^9 - 1 = 511
         if (_expectedSystemContractUpgradeTxHash == bytes32(0)) {
-            if (processedLogs != 127) {
-                revert MissingSystemLogs(127, processedLogs);
+            if (processedLogs != 255) {
+                revert MissingSystemLogs(255, processedLogs);
             }
-        } else if (processedLogs != 255) {
-            revert MissingSystemLogs(255, processedLogs);
+        } else if (processedLogs != 511) {
+            revert MissingSystemLogs(511, processedLogs);
+        }
+    }
+
+    /// @notice Precommits the status of all L2 transactions for the next batch on the shared bridge.
+    /// @param _batchNumber The sequential batch number to precommit (must equal `s.totalBatchesCommitted + 1`).
+    /// @param _precommitData ABI‐encoded transaction status list for the precommit.
+    function precommitSharedBridge(
+        uint256, // _chainId
+        uint256 _batchNumber,
+        bytes calldata _precommitData
+    ) external nonReentrant onlyValidator onlySettlementLayer {
+        uint256 expectedBatchNumber = s.totalBatchesCommitted + 1;
+        if (_batchNumber != expectedBatchNumber) {
+            revert InvalidBatchNumber(_batchNumber, expectedBatchNumber);
+        }
+        PrecommitInfo memory info = BatchDecoder.decodeAndCheckPrecommitData(_precommitData);
+        if (info.txs.length == 0) {
+            revert EmptyPrecommitData(_batchNumber);
+        }
+
+        bytes32 currentPrecommitment = s.precommitmentForTheLatestBatch;
+
+        uint256 length = info.txs.length;
+        for (uint256 i = 0; i < length; ++i) {
+            // todo: can optimize via assembly
+            bytes32 txStatusCommitment = keccak256(abi.encode(info.txs[i]));
+            currentPrecommitment = keccak256(abi.encode(currentPrecommitment, txStatusCommitment));
         }
+
+        s.precommitmentForTheLatestBatch = currentPrecommitment;
```

Comment:
could we name it `rollingTxsHash` in this way we keep the meaning, but detach it from precommit operation